node/src/main/scala/com/wavesplatform/database/RocksDBWriter.scala

```diff
@@ -680,6 +735,206 @@ class RocksDBWriter(
     log.trace(s"Finished persisting block ${blockMeta.id} at height $height")
   }
 
+  @volatile private var lastCleanupHeight = writableDB.get(Keys.lastCleanupHeight)
+  private def runCleanupTask(newLastSafeHeightForDeletion: Int, cleanupInterval: Int): Unit =
+    if (lastCleanupHeight + cleanupInterval < newLastSafeHeightForDeletion) {
+      cleanupExecutorService.submit(new Runnable {
+        override def run(): Unit = {
+          val firstDirtyHeight  = Height(lastCleanupHeight + 1)
+          val toHeightExclusive = Height(firstDirtyHeight + cleanupInterval)
+          val startTs           = System.nanoTime()
+
+          rdb.db.withOptions { (ro, wo) =>
+            rdb.db.readWriteWithOptions(ro, wo.setLowPri(true)) { rw =>
+              batchCleanupWavesBalances(
+                fromInclusive = firstDirtyHeight,
+                toExclusive = toHeightExclusive,
+                rw = rw
+              )
+
+              batchCleanupAssetBalances(
+                fromInclusive = firstDirtyHeight,
+                toExclusive = toHeightExclusive,
+                rw = rw
+              )
+
+              batchCleanupAccountData(
+                fromInclusive = firstDirtyHeight,
+                toExclusive = toHeightExclusive,
+                rw = rw
+              )
+
+              lastCleanupHeight = Height(toHeightExclusive - 1)
+              rw.put(Keys.lastCleanupHeight, lastCleanupHeight)
+            }
+          }
+
+          log.debug(s"Cleanup in [$firstDirtyHeight; $toHeightExclusive) took ${(System.nanoTime() - startTs) / 1_000_000}ms")
+        }
+      })
+    }
+
+  private def batchCleanupWavesBalances(fromInclusive: Height, toExclusive: Height, rw: RW): Unit = {
+    val lastUpdateAt = mutable.LongMap.empty[Height]
+
+    val updateAt     = new ArrayBuffer[(AddressId, Height)]() // AddressId -> First height of update in this range
+    val updateAtKeys = new ArrayBuffer[Key[BalanceNode]]()
+
+    val changedKeyPrefix = KeyTags.ChangedWavesBalances.prefixBytes
+    val changedFromKey   = Keys.changedWavesBalances(fromInclusive) // Doesn't matter, we need this only to parse
```

Comment:
what does this comment mean?
`changedFromKey` is used not only for parsing but also in `iterateOverWithSeek` as "seek" parameter

Reply:
It means, `fromInclusive` doesn't affect the parsing result. Changed the comment to make it clear.